server/master/src/main/java/org/apache/accumulo/master/Master.java

```diff
@@ -459,8 +464,11 @@ private void upgradeZookeeper() {
           perm.grantNamespacePermission(user, Namespaces.ACCUMULO_NAMESPACE_ID,
               NamespacePermission.READ);
         }
-        perm.grantNamespacePermission("root", Namespaces.ACCUMULO_NAMESPACE_ID,
-            NamespacePermission.ALTER_TABLE);
+        // because we need to refer to the root username, we can't use the
+        // ZKPermHandler directly. The security object should work since
```

Comment:
```suggestion
        // ZKPermHandler directly (it won't work for PermHandlers which use ZKPermHandler, but
        // change the root user name in some way. Instead, the security object should work since
```

Reply:
```suggestion
        // ZKPermHandler directly since that violates our earlier assumption that we don't
        // care about contents of the username. PermissionHandlers that need to
        // encode the username in some way, i.e. the KerberosPermissionHandler, things would
        // fail. Instead we should be able to use the security object since
```